src/vm/genericdict.cpp

```diff
@@ -801,6 +801,10 @@ Dictionary::PopulateEntry(
                 _ASSERTE(nonExpansive);
                 return NULL;
             }
+            if(th.IsArray())
```

Comment:
This should be done by encoding these slots using `ELEMENT_TYPE_NATIVE_ARRAY_TEMPLATE_ZAPSIG` instead.

Other places in the system - like casting helpers - still expect that TypeDesc for Arrays - blindly converting the TypeDesc to MethodTable here would break them.